### ISSUE TITLE
predicted cluster number always be 2

### ISSUE THREAD

**Issue by kli017** [wq2012/SpectralCluster#47]
Hello, I was using the spectral clustering slove the inter-block label permutation problem of eend speaker diarization method. 
I initial the spe_clusterer with the default setting as below. And found it works well for the two speaker conversations. However, when the real speaker number greater than 2, the cluster still predict with 2 speaker even I set the max_clusters as 7. I also tried with some different settings but it failed to predict the speaker number at most time. While the AHC is to sensetive with the distance hreshold and predicted more speaker than the real condition. I was wondering which paramter should I tune to make the scp_clusterer more sensetive with the speakr number? Thanks a lot.
```
scp_clusterer = SpectralClusterer(
        min_clusters=2,
        max_clusters=7,
        autotune=None,
        laplacian_type=None,
        refinement_options=None,
        custom_dist="cosine")
```

**Comment by wq2012** [comment]
What you are using is the vanilla spectral clustering without any refinement. Refinement is critical for more accurate estimation of speaker count. You can use these configs as examples: https://github.com/wq2012/SpectralCluster/blob/master/spectralcluster/configs.py

**Comment by kli017** [comment]
Thanks for the remind. I tried the turntodiarize cluster in the configs. I noticed it need a constrain matirx computed by turn score vector while doing predict. However I do not have that score. Instead I computed the distance matirx of the embddings and  set the CL and ML according to the distance.  But I still just get 2 speaker as result. If I use the distance matrix direclty I can get multiple speaker, however I found the result does not good. I was wondering how should I get a constrain matirx properly?
Thanks.
